### PR TITLE
Tpfa temp

### PR DESCRIPTION
**Pull request by hnil**
-- set tpfa for energy simulation.
-- disable diffusion for energy (pullrequest will come to fix this for tpfa)
-- added checking for running case static definitions

this PR depends on https://github.com/OPM/opm-models/pull/822

### CODE REVIEW

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -789,6 +791,25 @@ public:
         return *pffDofData_.get(context.element(), toDofLocalIdx).diffusivity;
     }
 
+    /*!
+     * give the transissibility for a face i.e. pair. should be symmetic?
```
Comment (akva2) [inline ebos/eclproblem.hh:795]: transmissibility, symmetric

File: ebos/eclcpgridvanguard.hh [hunk 2]
```diff
@@ -108,9 +109,73 @@ public:
...
+    void checkConsistency()
+    {
+        const auto& runspec = this->eclState().runspec();
+        const auto& config = this->eclState().getSimulationConfig();
+        const auto& phases = runspec.phases();
+
+        // check for correct module setup
+        if (config.isThermal()) {
+            if (getPropValue<TypeTag, Properties::EnableEnergy>() == false) {
+                throw std::runtime_error("Input spesify energy while simulator has disabled try xxx_energy");
```
Comment (akva2) [inline ebos/eclcpgridvanguard.hh:128]: specify. Comma before try. More of these below.

**Review by akva2 — APPROVED** [review]
Code changes look fine. Some typos spotted.

File: flow/flow_ebos_gasoil_energy.cpp [hunk 3]
```diff
@@ -59,6 +61,15 @@ template<class TypeTag>
...
     static constexpr bool value = true;
 };
+template<class TypeTag>
+struct Linearizer<TypeTag, TTag::EclFlowGasOilEnergyProblem> { using type = TpfaLinearizer<TypeTag>; };
+
+template<class TypeTag>
+struct LocalResidual<TypeTag, TTag::EclFlowGasOilEnergyProblem> { using type = BlackOilLocalResidualTPFA<TypeTag>; };
+
+template<class TypeTag>
+struct EnableDiffusion<TypeTag, TTag::EclFlowGasOilEnergyProblem> { static constexpr bool value = false; };
```
[the remark this answers is not in the document]
Comment (GitPaean) [reply flow/flow_ebos_gasoil_energy.cpp:71]: It looks like only solution is to introduce another flow variant to handle the situation with `DIFFUSE` activated.

[the remark this answers is not in the document]
Comment (GitPaean) [reply flow/flow_ebos_gasoil_energy.cpp:71]: With https://github.com/OPM/opm-simulators/pull/4825 getting in, this problem has been fixed.

**Pull request merged**